cli/internal/plugins/versions.go

```diff
@@ -61,11 +63,14 @@ func getLatestCQPluginRelease(ctx context.Context, name string, typ PluginType)
 	return extractVersionFromTag(mr.Latest), nil
 }
 
-// extractVersionFromTag takes a tag of the form "plugins-source-test-v0.1.21" and returns
-// the version, i.e. "v0.1.21"
+// extractVersionFromTag takes a tag of the form "plugins-source-test-v0.1.21" or "cli-v1.1.0-pre.1" and returns
+// the version, i.e. "v0.1.21" or "v1.1.0-pre.1"
 func extractVersionFromTag(tag string) string {
-	parts := strings.Split(tag, "-")
-	return parts[len(parts)-1]
+	m := reVersionFromTag.FindStringSubmatch(tag)
```

Comment:
isn't our versioning and tags are predictable and according to configured format in goreleaser? why do we need a regex? it's usually much harder to understand, debug, slower and so on...

Reply:
@yevgenypats I'm not sure how we would do this reliably without a regex. Or I can think of a solution, but it's not going to be better IMO. We don't know how many dashes there are in the part before the version starts, so we'd have to do something like this (pseudocode):

```
parts = split(tag, "-")
if len(parts) > 1 and parts[-1] matches v{}.{}.{}:
   return parts[-1]
if len(parts) > 2 and parts[-2] matches v{}.{}.{}:
   return parts[-2] + "-" + parts[-1]
return ""
```

We'd still need a regex for the `matches v{}.{}.{}` part there--`hasPrefix("v")` wouldn't cut it as you could have a plugin name that starts with a `v`.

Reply:
@yevgenypats refactored in https://github.com/cloudquery/cloudquery/pull/2097/commits/7ba878310946745c0069356bca862270b62bbe3c, thanks for the suggestion 👍